### PR TITLE
Update the Upgrading Using a Direct Download guide section.

### PR DESCRIPTION
**Pull request by andsel**
**PREVIEW:** https://logstash_13684.docs-preview.app.elstc.co/guide/en/logstash/master/upgrading-using-direct-download.html

## Release notes

[rn:skip]

## What does this PR do?
During the time Logstash switched from a set of Ruby files plus one fat jar to multiple jars, with configuration files
and shell scripts that also needs to be updated. It's not anymore sufficient to suggest to unpack a new distribution
over an existing one. This is problematic when we add jars from version to version because drive to the pollution
of the classpath.

This commit redefines to steps to do this, in a safer way, backing up data and config folders.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: docs/static/upgrading.asciidoc [hunk 1]
```diff
@@ -77,11 +77,14 @@ This procedure downloads the relevant Logstash binaries directly from Elastic.
 
 1. Shut down your Logstash pipeline, including any inputs that send events to Logstash.
 2. Download the https://www.elastic.co/downloads/logstash[Logstash installation file] that matches your host environment.
```
Comment (karenzone) [inline docs/static/upgrading.asciidoc:79]:
```suggestion
. Shut down your Logstash pipeline, including any inputs that send events to Logstash.
. Download the https://www.elastic.co/downloads/logstash[Logstash installation file] that matches your host environment.
```

Comment (karenzone) [reply docs/static/upgrading.asciidoc:79]: This notation sets up autonumbering. :-)

File: docs/static/upgrading.asciidoc [hunk 2]
```diff
@@ -77,11 +77,14 @@ This procedure downloads the relevant Logstash binaries directly from Elastic.
...
-3. Unpack the installation file into your Logstash directory.
-4. Test your configuration file with the `logstash --config.test_and_exit -f <configuration-file>` command. 
+3. Backup your `config/` and `data/` folders in a temporary space.
+4. Remove your Logstash directory.
+5. Unpack the installation file into the same folder your existing Logstash directory was present.
+6. Restore the `config/` and `data/` folders that was previously saved, overwriting the folders created during the unpack operation.
+7. Test your configuration file with the `logstash --config.test_and_exit -f <configuration-file>` command.
```
Comment (karenzone) [inline docs/static/upgrading.asciidoc:84]:
```suggestion
. Backup your `config/` and `data/` folders in a temporary space.
. Delete your Logstash directory.
. Unpack the installation file into the folder that contained the Logstash directory that you just deleted.
. Restore the `config/` and `data/` folders that were previously saved, overwriting the folders created during the unpack operation.
. Test your configuration file with the `logstash --config.test_and_exit -f <configuration-file>` command.
```

File: docs/static/upgrading.asciidoc [hunk 3]
```diff
@@ -77,11 +77,14 @@ This procedure downloads the relevant Logstash binaries directly from Elastic.
...
 Configuration options for
 some Logstash plugins have changed in the 7.x release.
-5. Restart your Logstash pipeline after updating your configuration file.
+8. Restart your Logstash pipeline after updating your configuration file.
```
Comment (karenzone) [inline docs/static/upgrading.asciidoc:87]:
```suggestion
. Restart your Logstash pipeline after updating your configuration file.
```

**Review by karenzone — COMMENTED** [review]
Minor formatting and wording suggestions for your consideration. 
FYI: I usually try to validate procedures as part of a review. I did not test these steps.

**Comment by andsel** [comment]
Thanks @karenzone
I've integrated you suggestion and executed the steps upgrading a `7.4`to a `7.16` and a `6.8` to `7.16`, the steps listed were ok.
The only doubt I have, which is nor covered by the other update methods, is that if a user previously installed plugins that's not part of the distribution, he has to remember to manually reinstall those.

File: docs/static/upgrading.asciidoc [hunk 4]
```diff
@@ -75,13 +75,16 @@ some Logstash plugins have changed in the 7.x release.
...
-4. Test your configuration file with the `logstash --config.test_and_exit -f <configuration-file>` command. 
+. Shut down your Logstash pipeline, including any inputs that send events to Logstash.
+. Download the https://www.elastic.co/downloads/logstash[Logstash installation file] that matches your host environment.
+. Backup your `config/` and `data/` folders in a temporary space.
+. Delete your Logstash directory.
+. Unpack the installation file into the folder that contained the Logstash directory that you just deleted.
+. Restore the `config/` and `data/` folders that were previously saved, overwriting the folders created during the unpack operation.
+. Test your configuration file with the `logstash --config.test_and_exit -f <configuration-file>` command.
 Configuration options for
 some Logstash plugins have changed in the 7.x release.
```
Comment (karenzone) [inline docs/static/upgrading.asciidoc:86]:
```suggestion
some Logstash plugins have changed.
```

Comment (karenzone) [reply docs/static/upgrading.asciidoc:86]: What do you think about this suggestion, @andsel?  It seems like some plugins changed in 7.x and others changed between 7.x and 8.0.

**Review by karenzone — APPROVED** [review]
I left one comment inline. Otherwise, builds cleanly and LGTM. Thanks for your work on this.

File: docs/static/upgrading.asciidoc (hunk 4 above)
Comment (andsel) [reply docs/static/upgrading.asciidoc:86]: correct, we can drop that part of the sentence.

**Pull request merged**